### PR TITLE
Fix #426 by not deleting the whole wsuser to do an update

### LINKED ISSUE #426: Duplicate on insertion into wsuser
I see a [number of errors inserting into wsuser](https://www.warmshowers.org/admin/reports/logs_filterable?uid=&message=&type%5B%5D=php&timestamp%5Bmin%5D=&timestamp%5Bmax%5D=&hostname=), for example 

```
Duplicate entry &#039;63647&#039; for key &#039;PRIMARY&#039; query: insert into wsuser (uid,fullname ,comments ,notcurrentlyavailable ,becomeavailable ,homephone ,mobilephone ,workphone ,fax_number ,preferred_notice ,maxcyclists ,motel ,campground ,bikeshop ,languagesspoken ,URL ,bed ,food ,laundry ,lawnspace ,sag ,shower ,storage ,kitchenuse ,howdidyouhear ,set_unavailable_timestamp ,set_available_timestamp ,last_unavailability_pester ) values ( 63647,&#039;Hamilton Fast&#039;,&#039;I am an avid cyclist and setting off on my first major cycling tour this fall. Enjoy meeting new people and experiencing new cultures. I am a courteous house guest and have no allergies or food restrictions. Other interests include anything to do with the outdoors, ultimate frisbee and photography. &#039;,&#039;1&#039;,1460354400,&#039;&#039;,&#039;&#039;,&#039;&#039;,&#039;&#039;,&#039;&#039;,&#039;1&#039;,&#039;&#039;,&#039;&#039;,&#039;&#039;,&#039;&#039;,&#039;&#039;,&#039;0&#039;,&#039;0&#039;,&#039;0&#039;,&#039;0&#039;,&#039;0&#039;,&#039;0&#039;,&#039;0&#039;,&#039;0&#039;,&#039;&#039;,0,1411396039,0 ) in /var/www/warmshowers.org/docroot/sites/all/modules/dev/wsuser/wsuser.module on line 1216
```

Location: https://github.com/warmshowers/Warmshowers.org/blob/site/docroot/sites/all/modules/dev/wsuser/wsuser.module#L1216-1216

### PR DESCRIPTION
**Pull request by rfay**
This looks like the right approach to this to me. It keeps the old meta-definition of fields, but adds what's needed to construct a proper INSERT/UPDATE statement, instead of the delete+add that it was doing (from what year of dev? yuck!)

Note that the equivalent in user_location does *not* do this delete + insert thing, so should not suffer the same fate. And it also does what it should by setting the item to NULL.

### CODE REVIEW

**Comment by rfay** [comment]
@kostajh, @illmasterc if you have just a few minutes to review this it would be appreciated. I think it's a long-delayed improvement (can't believe that code survived here that long). But it's also quite sensitive and could mess up a lot of things. 

The problem was that the mobile apps (both) seem to sometimes do rapid changes (login attempts mostly) via services, causing the wsuser row to get deleted and re-added (in the old version, stupid) and with the repeats, we get both errors and empty damaged wsuser rows. #426 has the full info.

I'll probably try to deploy it today, 1/29/2015

File: docroot/sites/all/modules/dev/wsuser/wsuser.module [hunk 1]
```diff
@@ -944,38 +945,45 @@ function wsuser_fieldlist() {
...
     'workphone' => array(
       'type' => 'textfield',
       'title' => t("Work Phone Number"),
+      'placeholder' => '"%s"',
     ),
-    'fax_number' => array(
-      'type' => 'textfield',
-      'title' => t("Fax Number"),
-    ),
+//    'fax_number' => array(
```
Comment (kostajh) [inline docroot/sites/all/modules/dev/wsuser/wsuser.module]: Oh no @rfay, I only accept hosting requests via :fax:! j/k

**Comment by rfay** [comment]
And here I wondered if anybody would ever notice the fax line's disappearance! I was WRONG!

**Comment by illmasterc** [comment]
Hmm. It seems like the right approach. I don't really have time to look into whether or not `function wsuser_save` has other calls throughout the code with 3 parameters, as it's now dropping to two, or whether this actually works as intended. If you're ready to deploy though, I trust you @rfay ! :smile_cat:

**Comment by rfay** [comment]
Thanks both - Yes, wsuser_save() is just called the one place. Did learn something amazing about D6 in 4bed00f though debugging, there was fallout!

**Pull request merged**